File: signature/jquery_cookie.go
package signature

var JQueryCookieSignature = Signature{
	Name:        "jQuery Cookie",
	Description: "A simple, lightweight jQuery plugin for reading, writing and deleting cookies.",
	Cpe:         "cpe:/a:jquery.cookie_project:jquery.cookie",

	DetectPattern: DetectPattern{
		URLs: []string{
			"jquery-cookie",
		},
		Bodies: []string{
			"jQuery Cookie Plugin",
		},
	},
	VersionPattern: VersionPattern{
		URLs: []string{
			`jquery-cookie[@/](\d+\.\d+\.\d+)`,
		},
		Bodies: []string{
			`jQuery Cookie Plugin v(\d+\.\d+)`,
		},
	},
}

File: src/signatures/technologies/jquery_cookie.ts
import type { Signature } from "../_types.js";
import { jquerySignature } from "./jquery.js";

export const jqueryCookieSignature: Signature = {
  name: "jQuery Cookie",
  description:
    "A simple, lightweight jQuery plugin for reading, writing and deleting cookies.",
  cpe: "cpe:/a:jquery.cookie_project:jquery.cookie",
  rule: {
    confidence: "high",
    urls: ["jquery-cookie/(\\d+\\.\\d+\\.\\d+)?"],
  },
  impliedSoftwares: [jquerySignature.name],
};
